module/zfs/zvol.c

```diff
@@ -1573,10 +1613,16 @@ zvol_create_minors_impl(const char *name)
 	int error = 0;
 	fstrans_cookie_t cookie;
 	char *atp, *parent;
+	list_t minors_list;
```

Comment:
I agree.  A good place to do this would be above the `list_create(&minors_list, ...)` a few lines down.  I'd also make a point of mentioning that it's safe to add/remove entries from this list without any locking because it's never modified by the taskq function.